src/groupByKey.ts

```diff
@@ -2,6 +2,16 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  const result: GroupsMap<T> = {};
+
+  items.forEach((item) => {
+    const itemKey = String([item[key]]);
+
+    const filtredItems = items.filter((elem) => String(elem[key]) === itemKey);
+
+    result[itemKey] = filtredItems;
```

Comment:
why do you need filter here?
you just need to check, if this itemKey property is already exist in you result, just push current item into it, or if it diesn't exist, create new array, and put your item into it